Src/AutoNSubstituteUnitTest/SubstituteAttributeRelayTest.cs

```diff
@@ -0,0 +1,163 @@
+﻿using System;
+using System.Reflection;
+using NSubstitute;
+using Ploeh.AutoFixture.Kernel;
+using Ploeh.TestTypeFoundation;
+using Xunit;
+
+namespace Ploeh.AutoFixture.AutoNSubstitute.UnitTest
+{
+    public class SubstituteAttributeRelayTest
+    {
+        [Fact]
+        public void SutIsSpecimenBuilderToServeAsFixtureCustomization()
+        {
+            // Fixture setup
+            // Exercise system
+            // Verify outcome
+            Assert.True(typeof(ISpecimenBuilder).IsAssignableFrom(typeof(SubstituteAttributeRelay)));
+            // Teardown
+        }
+
+        [Fact]
+        public void CreateReturnsNoSpecimenWhenRequestIsNull()
+        {
+            // Fixture setup
+            var sut = new SubstituteAttributeRelay();
+            var context = Substitute.For<ISpecimenContext>();
+            // Exercise system
+            object specimen = sut.Create(null, context);
+            // Verify outcome
+            Assert.IsType<NoSpecimen>(specimen);
+            // Teardown
+        }
+
+        [Fact]
+        public void CreateReturnsNoSpecimenWhenRequestIsNotICustomAttributeProvider()
+        {
+            // Fixture setup
+            var sut = new SubstituteAttributeRelay();
+            var request = new object();
+            var context = Substitute.For<ISpecimenContext>();
+            // Exercise system
+            object specimen = sut.Create(request, context);
+            // Verify outcome
+            var noSpecimen = Assert.IsType<NoSpecimen>(specimen);
+            Assert.Same(request, noSpecimen.Request);

```

Comment:
`NoSpecimen` has structural equality, so it'd be easier to read the assertion if it was expressed like this:

```C#
var expected = new NoSpecimen(request);
Assert.Equal(expected, specimen);
```